cirq-core/cirq/sim/simulator_test.py

```diff
@@ -101,6 +101,32 @@ def test_run_simulator_sweeps():
     assert simulator._run.call_count == 2
 
 
+@mock.patch.multiple(cirq.SimulatesSamples, __abstractmethods__=set(), _run=mock.Mock())
+def test_run_simulator_sweeps_with_deprecated_run():
+    simulator = cirq.SimulatesSamples()
```

Comment:
How hard would it be to create an actual simulator subclass for this test that has the deprecated `_run` behavior? Mocking always feel rather brittle to me.

Reply:
This test is just a copy of `test_run_simulator_sweeps` with `_run` modified to return a 2D array, but it's not particularly expensive to remove mocking throughout this file - especially since Cirq on the whole seems to have moved away from mocking since this was written. I'll make the change.

Reply:
Removed some of the mocks. The intermediate state mocks are a fair bit more convoluted, so I've left them for now.